client/homebrew/editor/editor.jsx

```diff
@@ -102,6 +104,50 @@ const Editor = createClass({
 		}, 1);
 	},
 
+	addEditorPageNumbers : function(){
+		if(!this.refs.codeEditor) return;
+		if(this.state.view === 'text')  {
+			const codeMirror = this.refs.codeEditor.codeMirror;
+
+			const pageCountWidgets = document.getElementsByClassName('editor-page-count');
+			for (let x=pageCountWidgets.length - 1;x>=0;x--) pageCountWidgets[x].remove();
+
+			let editorPageCount = 2; // start counting from page 2
+
+			const lineNumbers = _.reduce(this.props.brew.text.split('\n'), (r, line, lineNumber)=>{
+				if(this.props.renderer == 'legacy') {
+					if(line.includes('\\page')) {
+						const testElement = Object.assign(document.createElement('div'), {
+							className   : 'editor-page-count',
+							textContent : editorPageCount
+						});
+						codeMirror.addWidget({ line: lineNumber, ch: 0 }, testElement);
+						testElement.style.top = `${parseInt(testElement.style.top) - 12.5}px`;
+						testElement.style.left = 'unset';
```

Comment:
The div that is created is absolutely positioned and the `top` property is set by codemirror's addWidget method.  So each widget gets an inline-style for `top` and `left`.....which are based on the line's {line: , ch: } location in the editor.  By default, codemirror places the widget in the "next line", just below where we want it.

Since the `top` and `left` properties are set as inline styles, I think using an external stylesheet won't have any effect. 

Same with the other comment.

Reply:
I think this is a case that we can justify using `!important`, if it helps keep the styling all in one place instead of scattered between multiple documents.

How about in the .less file: `left : unset !important, top: transform: translateY(-12.5px)`

Reply:
Huh, that's obnoxious.

Well then, I think that this is a case where we can justify using `!important`, if it helps keep the styling all in one place instead of scattered between multiple documents.

How about in the .less file: `left : unset !important, top: transform: translateY(-12.5px)`

Reply:
What about for `left` just removing that property from the inline style attribute?  So it won't even show inline, and then we don't need to unset it in the stylesheet, and in the future it's easy to change to `left` for whatever reason.

And regarding `top`, my gut feeling is to leave it inline since that is the output of addWidget().  Rather than having the top position set in a single spot, it would be set in *two* spots.  And transforms skeeve me out for some reason.  However, my 'gut' doesn't have very many strong feelings so I'm amenable to changing per your recommendation, but wanted to give a chance for further reflection on it.

Reply:
> What about for left just removing that property from the inline style attribute?

Hmmmm.... that might be ok. Alright, if you fully remove `left` we can keep `top` here for now. But if possible can we change the magic 12.5px number to something that respects the actual line size in case somebody wants to zoom in and out or do something funky with the appearance of the editor? I wonder if `cm.defaultTextHeight()` or something similar would work?

Reply:
removed `left` entirely, and used `cm.defaulTextHeight()` to offset position of the widget.